src/Illuminate/Support/LazyCollection.php

```diff
@@ -0,0 +1,1441 @@
+
+                $values->next();
+            }
+
+            if ($values->valid()) {
+                trigger_error($errorMessage, E_USER_WARNING);
+            }
+        });
+    }
+
+    /**
+     * Union the collection with the given items.
+     *
+     * @param  mixed  $items
+     * @return static
+     */
+    public function union($items)
+    {
+        return $this->passthru('union', func_get_args());
+    }
+
+    /**
+     * Create a new collection consisting of every n-th element.
+     *
+     * @param  int  $step
+     * @param  int  $offset
+     * @return static
+     */
+    public function nth($step, $offset = 0)
+    {
+        $original = clone $this;
+
+        return new static(function () use ($original, $step, $offset) {
+            $position = 0;
+
+            foreach ($original as $item) {
+                if ($position % $step === $offset) {
+                    yield $item;
+                }
+
+                $position++;
+            }
+        });
+    }
+
+    /**
+     * Get the items with the specified keys.
+     *
+     * @param  mixed  $keys
+     * @return static
+     */
+    public function only($keys)
+    {
+        if ($keys instanceof Enumerable) {
+            $keys = $keys->all();
+        } elseif (! is_null($keys)) {
+            $keys = is_array($keys) ? $keys : func_get_args();
+        }
+
+        $original = clone $this;
+
+        return new static(function () use ($original, $keys) {
+            if (is_null($keys)) {
+                yield from $original;
+            } else {
+                $keys = array_flip($keys);
+
+                foreach ($original as $key => $value) {
+                    if (array_key_exists($key, $keys)) {
+                        yield $key => $value;
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * Get and remove the last item from the collection.
+     *
+     * @return mixed
+     */
+    public function pop()
+    {
+        $items = $this->collect();
+
+        $result = $items->pop();
+
+        $this->source = $items;
+
+        return $result;
+    }
+
+    /**
+     * Push an item onto the beginning of the collection.
+     *
+     * @param  mixed  $value
+     * @param  mixed  $key
+     * @return $this
+     */
+    public function prepend($value, $key = null)
+    {
+        $original = clone $this;
+
+        $this->source = function () use ($original, $value, $key) {
+            $instance = new static(function () use ($original, $value, $key) {
+                yield $key => $value;
+
+                yield from $original;
+            });
+
+            if (is_null($key)) {
+                $instance = $instance->values();
+            }
+
+            yield from $instance;
+        };
+
+        return $this;
+    }
+
+    /**
+     * Push an item onto the end.
+     *
+     * @param  mixed  $value
+     * @return $this
+     */
+    public function push($value)
```

Comment:
This method does the same as ::add(), maybe you should proxy one with the other.